### PR TITLE
ImplementedLoginAndAddedToSwagger

### PR DESCRIPTION
**Pull request by shumYevh**

### CODE REVIEW

File: src/main/java/org/example/bookweb/controller/BookController.java [hunk 1]
```diff
@@ -67,7 +67,8 @@ public void delete(@PathVariable Long id) {
 
     @GetMapping("/search")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/controller/BookController.java:68]: Should we add `@PreAuthorize` here?

File: src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getBearerToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```
use constant from Spring

File: src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java]:
```suggestion
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_HEADER)) {
```

private static final String TOKEN_HEADER = "Bearer ";

File: src/main/java/org/example/bookweb/security/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.example.bookweb.dto.user.UserLoginResponseDto;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/AuthenticationService.java]:
```suggestion
```

File: src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/org/example/bookweb/security/JwtUtil.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.Date;
+import java.util.function.Function;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/JwtUtil.java]:
```suggestion
```

File: src/main/java/org/example/bookweb/security/JwtUtil.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    private <T> T getClaimFromToken(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = Jwts.parserBuilder()
+                .setSigningKey(secret)
+                .build()
+                .parseClaimsJws(token)
+                .getBody();
+        return claimsResolver.apply(claims);
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/security/JwtUtil.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/org/example/bookweb/controller/BookController.java (hunk 1 above)
Comment (shumYevh) [reply src/main/java/org/example/bookweb/controller/BookController.java:68]: I am not sure

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**